src/DependencyGraph.php

```diff
@@ -100,7 +100,7 @@ public function findRootNodes()
     /**
      * Resolve this dependency graph. In the end a valid path will be returned.
      *
-     * @return DependencyNode[]
+     * @return string[]
```

Comment:
Good catch! The type is indeed incorrect, but the return type depends on the type that is passed into `DependencyNode` for the element, because the elements are returned. Right now `$element` is typed as `mixed` and so this would have to be 
```suggestion
     * @return mixed[]
```

I think it would be possible to solve that in a better way with generics.